### DISCUSSION TITLE
Can't map data with "string" data type

### DISCUSSION THREAD

**Discussion by onlylove855** [AutomaApp/automa#1238] [Q&A]
I want map data like this:
var aaa = "QSM"
map "QSM" to "ZZZ"
![image](https://user-images.githubusercontent.com/22310875/231642776-26e53b3d-0a5c-41d6-8546-12b1d32db93c.png)

when I execute it, it shows error "Can't map data with "string" data type"
![image](https://user-images.githubusercontent.com/22310875/231643030-01766dcb-a488-4430-a368-90c4c9f6efc8.png)

Why am I getting this error?

**Comment by FooJiaYin** [comment]
+1 Also getting this error when mapping data

**Comment by sparkls0** [comment]
You're getting this error because your variable 'aaa' is a string type variable 

Map is a function for arrays

You need to turn your string into an array

**Reply by amaillet2024** [reply]
hello, how can we turn a string into an array ?

**Reply by amaillet2024** [reply]
found my own answer so I share it : 
create a insertData block (since JS doesn't work) and use push to create an array or add a value to an array. 
![2024-04-01 11_54_22-geev workflow - Automa](https://github.com/AutomaApp/automa/assets/163475905/4055358b-3f64-49f1-9a01-955cf6585353)
